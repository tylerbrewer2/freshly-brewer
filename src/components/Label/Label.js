import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Style Helpers
import Fonts from '../../styles/helpers/Fonts';

const LabelCore= ({ children, identifier, className }) => {
  return (
    <label
      className={className}
      htmlFor={identifier}
    >
      {children}
    </label>
  )
}

LabelCore.propTypes = {
  children: PropTypes.string.isRequired,
  indentifier: PropTypes.string.isRequired,
}

const Label = styled(LabelCore)`
  font-family: ${Fonts.body};
  font-size: 18px;
  text-align: left;
  display: block;
  margin-bottom: 5px;
`

export default Label;
