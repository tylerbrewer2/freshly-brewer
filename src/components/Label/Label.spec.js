import React from 'react';
import renderer from 'react-test-renderer';

import Label from './Label';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Label identifier="identifier">
        Test value
      </Label>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
