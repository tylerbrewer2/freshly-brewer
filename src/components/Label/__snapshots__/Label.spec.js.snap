exports[`component structure renders correctly 1`] = `
<label
  className="sc-bdVaJa fxdtZh"
  for="identifier">
  Test value
</label>
`;
